app/services/activitypub/process_collection_service.rb

```diff
@@ -28,6 +28,14 @@ def different_actor?
     @json['actor'].present? && value_or_id(@json['actor']) != @account.uri
   end
 
+  def suspended_actor?
+    @account.suspended? && !(@account.suspension_origin_remote? && activity_allowed_while_suspended?)
+  end
+
+  def activity_allowed_while_suspended?
+    %w(Delete Reject Undo Update).include?(@json['type'])
```

Comment:
We probably want to allow deletion of content even for locally-suspended remote accounts here. Two reasons:
- suspending a remote account that then gets deleted on the remote server would make us miss the `Delete`
- we should let the remote suspended account still remove followers (we clear the follow relationships from the suspended account, not the other way around)